docs_src/microservices/security/edgex-config-proxy.1.md

```diff
@@ -0,0 +1,127 @@
+
+# COMMANDS
+
+  * **tls**
+
+    Configure inbound TLS certificate. This command will configure the TLS secrets into the secret store and re-deploy them to Kong. Requires additional arguments:
+
+    * --certificate _/path/to/certchain.pem_
+
+      Path to TLS leaf certificate (with optional appended certificate chain, not including a root certificate).
+
+    * --key _/path/to/private.pem_
+
+      Path to TLS private key.
+
+  * **adduser**
+
+    Create a proxy user using JWT public/private key authentication. Requires additional arguments:
+
+    * --jwt (required)
+
+      Create user using JWT authentication plugin.
+
+    * --algorithm HS256 | ES256 (required)
+
+      Algorithm used for signing the JWT.
+
+    * --public\_key _public\_key.pem_ (required)
+
+      Public key used to validate the JWT. This key is assumed to have been pre-created using some external mechanism such as a TPM, HSM, openssl, or other method.
+
+    * --user _username_ (required)
+
+      Username of the user to add.
+
+    * --id _key_
+
+      Optional user-specified &quot;key&quot; used for linkage with an incoming JWT via Kong&#39;s config.key\_claim\_name setting (defaults to &quot;iss&quot; field).
+
+    * --group _group_
+
+      Group to which the user belongs, defaults to &quot;admin&quot;.
+
+    * --output-key _path_
+
+      Output auto-genreated to a file (or &quot;-&quot; for stdout).
+
+  * **deluser**
+
+    Delete a proxy user. Requires additional arguments:
+
+    * --jwt (required)
+
+      Create user using JWT authentication plugin.
+
+  * **jwt**
+
+    Utility function to create a JWT proxy authentication token from a supplied secret. This command does not require secret store access, but the values supplied must match those presented to the adduser command earlier. Requires additional arguments:
+
+    * --algorithm `HS256` | `ES256` (required)
+
+      Algorithm used for signing the JWT.
+
+    * --id _key_ (required)
+
+      The &quot;key&quot; field from the &quot;adduser&quot; command.
+
+    * --private\_key _private\_key.pem_
+
+      Private key used to sign the JWT.
+
+    * --exp _duration_
+
+      Duration of generated jwt expressed as a golang-parseable duration value. Use &quot;never&quot; to omit an expiration field in the JWT. Defaults to &quot;1h&quot; (one hour) if unspecified.
+
+    * --output _path_
+
+      Output encoded JWT token to a file (or &quot;-&quot; for stdout).
+
+      The generated JWT will be the encoded representation of:
+
+      <pre>
+      {
+        &quot;typ&quot;: &quot;JWT&quot;,
+        &quot;alg&quot;: &quot;HS256 | ES256&quot;
+      }
+      {
+        &quot;iss&quot;: &quot;_key_&quot;,
+        &quot;exp&quot;: (calculated expiration time)
+      }
+      (signature)
+      </pre>
+
+# ENVIRONMENT
+
+  * **IKM\_HOOK**
+
+    Enables decryption of an encrypted secret store master key by pointing at an executable that returns an encryption seed.
```

Comment:
How? Example?

Reply:
These are the same parameters used by security-secretstore-setup for HW security hooks.